src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -793,10 +793,7 @@ public function processDeleteVirtualProduct($product)
     {
         $id_product_download = ProductDownload::getIdFromIdProduct((int) $product->id, false);
         $download = new ProductDownload($id_product_download ? $id_product_download : null);
-
-        if ($download) {
-            $download->delete(true);
-        }
+        $download->delete(true);
```

Comment:
Still check if the download object is properly loaded

Reply:
Fixed